src/pages/home/report/ContextMenu/PopoverReportActionContextMenu.js

```diff
@@ -278,69 +248,81 @@ class PopoverReportActionContextMenu extends React.Component {
+        reportActionRef.current = reportAction;
+
+        setShouldSetModalVisibilityForDeleteConfirmation(shouldSetModalVisibility);
+        setIsDeleteCommentConfirmModalVisible(true);
+    };
+
+    useImperativeHandle(ref, () => ({
+        showContextMenu,
+        hideContextMenu,
+        showDeleteModal,
+        hideDeleteModal,
+        isActiveReportAction,
+        instanceID,
+        runAndResetOnPopoverHide,
+    }));
+
+    const reportAction = reportActionRef.current;
+
+    return (
+        <>
+            <PopoverWithMeasuredContent
+                isVisible={isPopoverVisible}
+                onClose={hideContextMenu}
+                onModalShow={runAndResetOnPopoverShow}
+                onModalHide={runAndResetOnPopoverHide}
+                anchorPosition={popoverAnchorPosition.current}
+                animationIn="fadeIn"
+                disableAnimation={false}
+                animationOutTiming={1}
+                shouldSetModalVisibility={false}
+                fullscreen
+                withoutOverlay
+                anchorRef={anchorRef}
+            >
+                <BaseReportActionContextMenu
+                    isVisible
+                    type={typeRef.current}
+                    reportID={reportIDRef.current}
+                    reportAction={reportAction}
+                    draftMessage={reportActionDraftMessageRef.current}
+                    selection={selectionRef.current}
+                    isArchivedRoom={isArchivedRoom}
+                    isChronosReport={isChronosReport}
+                    isPinnedChat={isPinnedChat}
+                    isUnreadChat={isUnreadChat}
+                    anchor={contextMenuTargetNode}
+                    contentRef={contentRef}
                 />
-            </>
-        );
-    }
-}
+            </PopoverWithMeasuredContent>
+            <ConfirmModal
+                title={props.translate('reportActionContextMenu.deleteAction', {reportAction})}
+                isVisible={isDeleteCommentConfirmModalVisible}
+                shouldSetModalVisibility={shouldSetModalVisibilityForDeleteConfirmation}
+                onConfirm={confirmDeleteAndHideModal}
+                onCancel={hideDeleteModal}
+                onModalHide={() => {
+                    reportIDRef.current = '0';
+                    reportActionRef.current = {};
+                    callbackWhenDeleteModalHide.current();
+                }}
+                prompt={props.translate('reportActionContextMenu.deleteConfirmation', {action: reportAction})}
+                confirmText={props.translate('common.delete')}
+                cancelText={props.translate('common.cancel')}
+                danger
+            />
+        </>
+    );
+});
 
 PopoverReportActionContextMenu.propTypes = propTypes;
+PopoverReportActionContextMenu.displayName = 'PopoverReportActionContextMenu';
 
 export default withLocalize(PopoverReportActionContextMenu);
```

Comment:
remove the `withLocalize` and use the hook `useLocalize`

Reply:
done